sameinad.py

```diff
@@ -0,0 +1,401 @@
+#----------------------------------------------------------------------------------------------------
+# Setup and Environment Configuration
+#----------------------------------------------------------------------------------------------------
+
+from sqlalchemy import create_engine
+from shiny import App, render, ui
+import pandas as pd
+import geopandas as gpd
+from shapely import wkt
+import folium
+from folium.plugins import MarkerCluster
+from matplotlib import cm, colors
+import numpy as np
+import plotly.express as px
+from shinywidgets import output_widget, render_widget
+import os
+from dotenv import load_dotenv
+
+# Load environment variables from the .env file
+load_dotenv()
+
+# Fetch database credentials from environment variables
+db_user = os.getenv("DB_USER")
+db_password = os.getenv("DB_PASSWORD")
+db_host = os.getenv("DB_HOST")
+db_port = os.getenv("DB_PORT")
+db_name = os.getenv("DB_NAME")
+
+# Check if all variables are set
+if not all([db_user, db_password, db_host, db_port, db_name]):
+    raise ValueError("One or more environment variables are not set.")
+
+# Create the database engine
+engine = create_engine(f"postgresql://{db_user}:{db_password}@{db_host}:{db_port}/{db_name}")
+
+
+#----------------------------------------------------------------------------------------------------
+# Data Fetching and Processing
+#----------------------------------------------------------------------------------------------------
+
+# Function to fetch houses and kingdoms data with explicit mapping
+def fetch_houses_and_kingdoms():
+    query = """
+    SELECT 
+        h.name AS house_name,
+        CASE
+            WHEN h.region IN ('The North', 'The Neck', 'Beyond the Wall') THEN 'The North'
+            WHEN h.region = 'The Vale' THEN 'The Vale'
+            WHEN h.region = 'Iron Islands' THEN 'Iron Islands'
+            WHEN h.region = 'The Riverlands' THEN 'The Riverlands'
+            WHEN h.region = 'The Westerlands' THEN 'The Westerlands'
+            WHEN h.region = 'The Stormlands' THEN 'The Stormlands'
+            WHEN h.region = 'The Crownlands' THEN 'The Crownlands'
+            WHEN h.region = 'The Reach' THEN 'The Reach'
+            WHEN h.region = 'Dorne' THEN 'Dorne'
+            ELSE 'Other Regions'
```

Comment:
Þið mættuð einfalda þetta talsvert
```suggestion
            WHEN h.region IN ('The North', 'The Neck', 'Beyond the Wall') THEN 'The North'
            WHEN h.region is not null THEN h.region
            ELSE 'Other Regions'
```
en af hverju viljið þið sameina Norðið með 'The Neck' og 'Beyond the Wall' - þau eru óbyggileg svæði / utan siðmenningar